src/engine/LocalVocab.h

```diff
@@ -97,6 +97,9 @@ class LocalVocab {
   void mergeWith(const R& vocabs) {
     auto inserter = std::back_inserter(otherWordSets_);
     for (const auto& vocab : vocabs) {
+      if (vocab.empty()) {
+        continue;
+      }
```

Comment:
And you should also refactor the
LocalVocab::clone() method to call this `mergeWith` method:

``` 
... clone() {
LocalVocab clone;
clone.mergeWith(std::span(this, 1));
return clone;
}
```